docs/azure_jumpstart_ag/contoso_supermarket/ci_cd/_index.md

```diff
@@ -4,4 +4,238 @@ weight: 100
+
+- The repository has two main folders to separate the development team applications' code (developer) and the DevOps team operations GitOps configurations (operations).
+
+    ![Screenshot showing the GitHub repository main folder structure](./img/repo_folder_structure.png)
+
+- Within the _developer_ folder there is a folder for each application's source code. This is where Contoso Supermarket's developers develop new features.
+
+    ![Screenshot showing the developer folder structure](./img/repo_developer_structure.png)
+
+- Within the _operations_ folder there is also a folder for each application's GitOps configuration. This is where Contoso Supermarket's DevOps team manages how the applications are deployed to the different environments.
+
+    ![Screenshot showing the operations folder structure](./img/repo_operations_structure.png)
+
+- Each application has a folder for [Helm](https://helm.sh/docs/) _charts_ where the Kubernetes manifests for each application are located and a folder for _releases_ where the Helm Releases for each application and each environment is located. This way the DevOps team can control the promotion of each version of the applications across the CI/CD workflow on different environments and also enable/disable features created by the developers as needed.
+
+    ![Screenshot showing the helm folder structure](./img/repo_operations_helm_structure.png)
+
+    ![Screenshot showing the helm releases folder structure](./img/repo_operations_helmreleases_structure.png)
+
+- Helm Releases control the promotion of the PoS application images from one environment to the other using the image tag as a Helm value.
+
+    ![Screenshot showing the helm release values](./img/github_helm_release_example.png)
+
+- The repository has the following GitHub actions that build the Contoso Supermarket's CI/CD workflow
+  - **Promote-to-staging:** This workflow is triggered once a pull request targeting the code of the PoS application is merged. It will build the new images based on code changes, increment the image tag, run unit tests and update the PoS application on the _Staging_ environment.
+  - **Promote-to-canary:** This workflow is triggered once the staging cluster has a healthy deployment of the PoS application, this is done once the Flux v2 operator captures the new commit in staging branch and generates an notification that the deployment is ready. It will build the new canary image, run integration tests and submits a PR to update the image tag on the _canary_ Helm release.
+  - **Promote-to-production:** This workflow is triggered once the canary cluster has a healthy deployment of the PoS application, this is done once the Flux v2 operator captures the new commit in canary branch and generates an notification that the deployment is ready. It will build the new production image, run final tests and submits a PR to update the image tag on the _production_ Helm release.
```

Comment:
```suggestion
  - __Promote-to-staging:__ This workflow is triggered once a pull request targeting the code of the PoS application is merged. It will build the new images based on code changes, increment the image tag, run unit tests, and update the PoS application on the _Staging_ environment.
  - __Promote-to-canary:__ This workflow is triggered once the staging cluster has a healthy deployment of the PoS application, this is done once the Flux v2 operator captures the new commit in the staging branch and generates a notification that the deployment is ready. It will build the new canary image, run integration tests, and submits a PR to update the image tag on the _canary_ Helm release.
  - __Promote-to-production:__ This workflow is triggered once the canary cluster has a healthy deployment of the PoS application, this is done once the Flux v2 operator captures the new commit in the canary branch and generates a notification that the deployment is ready. It will build the new production image, run final tests, and submits a PR to update the image tag on the _production_ Helm release.
```